engine/src/main/scala/cromwell/engine/db/ExecutionInfosByExecution.scala

```diff
@@ -0,0 +1,107 @@
+package cromwell.engine.db
+
+import cromwell.engine._
+import cromwell.engine.backend._
+import cromwell.engine.db.slick.{Execution, ExecutionInfo}
+import wdl4s.values.WdlFile
+
+case class ExecutionInfosByExecution(execution: Execution, executionInfos: Seq[ExecutionInfo]) {
+
+  lazy val callLogs: Option[CallLogs] = {
+    import ExecutionInfosByExecution._
+
+    case class Accumulator(stdout: Option[WdlFile], stderr: Option[WdlFile], logs: Map[String, WdlFile])
+
+    val acc = executionInfos.foldLeft(Accumulator(None, None, Map.empty)) { (acc, executionInfo) =>
+      executionInfo.key.split("_", 2) match {
+        case Array(CallLogPrefix, StdoutSuffix) => acc.copy(stdout = executionInfo.value map {
+          WdlFile(_)
+        })
+        case Array(CallLogPrefix, StderrSuffix) => acc.copy(stderr = executionInfo.value map {
+          WdlFile(_)
+        })
+        case Array(CallLogPrefix, callLogName) if executionInfo.value.isDefined =>
+          acc.copy(logs = acc.logs + (callLogName -> WdlFile(executionInfo.value.get)))
+        case _ => acc
+      }
+    }
+
+    acc match {
+      case Accumulator(Some(stdout), None, _) =>
+        throw new IllegalArgumentException(s"stderr was not found for stdout $stdout")
+      case Accumulator(None, Some(stderr), _) =>
+        throw new IllegalArgumentException(s"stdout was not found for stderr $stderr")
+      case Accumulator(None, None, logs) if logs.nonEmpty =>
+        throw new IllegalArgumentException(s"stdout and stderr were empty logs $logs")
+      case _ =>
+        for {
+          stdout <- acc.stdout
+          stderr <- acc.stderr
+        } yield CallLogs(stdout, stderr, Option(acc.logs).filterNot(_.isEmpty))
+    }
+  }
+
+}
+
+object ExecutionInfosByExecution {
+  private val CallLogPrefix = "$log"
+  private val StdoutSuffix = "stdout"
+  private val StderrSuffix = "stderr"
+
+  def toCallLogMap(callLogs: CallLogs): Map[String, Option[String]] = {
```

Comment:
Is there a reason why this can't return a Map[String, String]?  Unless I'm reading this wrong, the value will always be a `Some`?